sdks/wasm/DebuggerTestSuite/Tests.cs

```diff
@@ -1243,6 +1247,141 @@ public async Task InspectLocalsForStructInstanceMethod (bool use_cfo)
+						Assert.Equal (0, loc.Count);
+						 						
+						//after Ready
+						await init_cmds[4];	
+						// Check that scripts have been loaded   
+						Assert.Contains (dicFileToUrl["dotnet://debugger-test.dll/debugger-test.cs"], event_order);
+
+						await EvaluateAndCheck (
+							"window.setTimeout(function() { invoke_add(); }, 1);",
+							"dotnet://debugger-test.dll/debugger-test.cs", 5, 2,
+							"IntAdd",
+							wait_for_event_fn: (pause_location) => {
+								var top_frame = pause_location ["callFrames"][0];
+								Assert.Equal ("IntAdd", top_frame ["functionName"].Value<string>());
+								return Task.CompletedTask;
+							}
+						);
+						Assert.Contains ("breakpointResolved", event_order);
+						var script_ix = event_order.IndexOf (dicFileToUrl["dotnet://debugger-test.dll/debugger-test.cs"]);
+						var resolve_ix = event_order.IndexOf ("breakpointResolved");
+						
+						// check breakpoint resolved after script is loaded
+						Assert.True (script_ix < resolve_ix);
+												
+					}, cts.Token);
+					await client.Close (cts.Token);
+				}
+			};
+		}
+
+		[Fact]
+		public async Task SetBreakpointAfterInspectorReady () {
+			var insp = new Inspector ();
+
+			var scripts = SubscribeToScripts (insp);
+			await Ready (); 
+			
+			using (var cts = new CancellationTokenSource()) {
+				cts.CancelAfter (60 * 1000);
+				var uri = new Uri ($"ws://{TestHarnessProxy.Endpoint.Authority}/launch-chrome-and-connect");
+				using var loggerFactory = LoggerFactory.Create(
+					builder => builder.AddConsole().AddFilter(null, LogLevel.Trace));
+				using (var client = new InspectorClient (loggerFactory.CreateLogger<Inspector>())) {
+					await client.Connect (uri, insp.OnMessage, async token => {
+						Task[] init_cmds = {
+							client.SendCommand ("Profiler.enable", null, token),
+							client.SendCommand ("Runtime.enable", null, token),
+							client.SendCommand ("Debugger.enable", null, token),
+							client.SendCommand ("Runtime.runIfWaitingForDebugger", null, token),
+							insp.WaitFor ("ready"),
+						};
+						
+						insp.On("Debugger.breakpointResolved", async (args, c) => { 
+							Console.WriteLine ("BREAKPOINT RESOLVED");
+							event_order.Add ("breakpointResolved");
+						});
+
+						Console.WriteLine ("waiting for the debugger to be ready");
+
+						// after inspector is ready
+						await init_cmds[4];
+
+						// check that script is loaded
+						Assert.Contains (dicFileToUrl["dotnet://debugger-test.dll/debugger-test.cs"], event_order);
+						
+						ctx = new DebugTestContext (client, insp, token, scripts);
+
+						// set breakpoint
+						var bp1_req = JObject.FromObject (new { lineNumber = 5, columnNumber = 2, url = dicFileToUrl["dotnet://debugger-test.dll/debugger-test.cs"],});
+						var bp1_res = await ctx.cli.SendCommand ("Debugger.setBreakpointByUrl", bp1_req, ctx.token);
+						Assert.True (true ? bp1_res.IsOk : bp1_res.IsErr);
```

Comment:
Why not `await SetBreakpoint ("dotnet://debugger-test.dll/debugger-test.cs", 5, 2, expect_ok: true)`?